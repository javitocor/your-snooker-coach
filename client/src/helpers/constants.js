export const DEV_URL = 'https://localhost:3000/api/v1';

export const GET_ALL_PLAYERS = 'GET_ALL_PLAYERS';
export const GET_ALL_PLAYERS_ERROR = 'GET__PLAYERS_ERROR';
export const GET_ALL_PLAYERS_PENDING = 'GET_ALL_PLAYERS_PENDING';
export const GET_PLAYER = 'GET_PLAYER';
export const GET_PLAYER_ERROR = 'GET_PLAYER_ERROR';
export const GET_PLAYER_PENDING = 'GET_PLAYER_PENDING';


export const initialStatePlayers = {
  error: null,
  pending: false,
  playersList: [],
  player: [],
}

export const GET_TRAININGS = 'GET_TRAININGS';
export const GET_TRAININGS_PENDING = 'GET_TRAININGS_PENDING';
export const GET_TRAININGS_ERROR = 'GET_TRAININGS_ERROR';
export const CREATE_TRAINING = 'CREATE_TRAINING';
export const CREATE_TRAINING_PENDING = 'CREATE_TRAINING_PENDING';
export const CREATE_TRAINING_ERROR = 'CREATE_TRAINING_ERROR';

