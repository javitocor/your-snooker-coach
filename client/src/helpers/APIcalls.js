import { DEV_URL } from './constants';

export const AllPlayersCall = () => async dispatch => {
  const Url = DEV_URL;
  try {
      dispatch(getAllPlayersPending());

    const response = await fetch(`${Url}/players`, { mode: 'cors' });
    const playersList = await response.json();
    dispatch(getAllPlayers(playersList));
    return playersList;
  } catch (error) {
      dispatch(getAllPlayersError(error));
  }
};

export const PlayerCall = (id) => async dispatch => {
  const Url = DEV_URL;
  try {
      dispatch(getPlayerPending());

    const response = await fetch(`${Url}/players/${id}`, { mode: 'cors' });
    const player = await response.json();
    dispatch(getPlayer(player));
    return player;
  } catch (error) {
      dispatch(getPlayerError(error));
  }
};