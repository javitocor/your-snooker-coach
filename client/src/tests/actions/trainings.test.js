import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { TrainingsCall } from '../../helpers/APIcalls';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
  let store;
  const training = [{
    date: '2022-03-05', location: 'EN', player_id: 1, user_id: 1,
  }];
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('creates getPlayer action', () => {
    nock('https://localhost:3000')
      .get('/api/v1/trainings')
      .reply(200, training);

    return store.dispatch(TrainingsCall())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
