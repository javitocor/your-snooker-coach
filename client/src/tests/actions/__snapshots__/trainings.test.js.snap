// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates getPlayer action 1`] = `
Array [
  Object {
    "type": "GET_TRAININGS_PENDING",
  },
  Object {
    "error": [TypeError: Network request failed],
    "type": "GET_TRAININGS_ERROR",
  },
]
`;
